mitodl/ocw-studio
# Convert more baseurl links to resource_links

#### Pre-Flight checklist


#### What are the relevant tickets?
This completes the second task in https://github.com/mitodl/ocw-studio/issues/1015

#### What's this PR do?
This PR does more baseurl --> resource_link conversion using the new LinkParser. In particular, we now convert about 950 more links that have square brackets in their link text or (non-consecutive) newlines. The earlier regex approach could not handle this.

#### How should this be manually tested?
1. **First, unescape some stuff:** Run `docker-compose run --rm web python manage.py markdown_cleanup link_unescape -o link_unescape.csv --commit` to unescape some link square brakcets that ought not to be escaped. Inspect the CSV to ensure replacements look reasonable.
2. **Now convert baseurls** Run `docker-compose run --rm web python manage.py markdown_cleanup baseurl -o baseurl.csv --csv-only-changes to generate a list of changes that will be made when the command is run in --comit mode.
3. Inspect list to makes sure it's reasonable / links work.
4. You may want to specifically check that the new resource links do NOT include newlines. Markdown allows non-consecutive newlines in the link text but does render them. Hugo shortcodes do NOT allow newlines in the shortcode arguments, as far as I can tell.
